### DISCUSSION TITLE
Authentication on subdomains and custom domains

### DISCUSSION THREAD

**Discussion by viperfx** [gladly-team/next-firebase-auth#409] [Q&A]
How can we use this library with a multi-tenant application that can have user created subdomains or if the application is accessed via a custom domain. 

site1.mydomain.com <- visit site1.mydomain.com/auth and authenticate. Cookies are stored for this domain and share with all subdomains on mydomain.com 

site2.customdomain.com <-> mapped to site2.mydomain.com
visit site2.custdomdomain.com/auth and authenticate. Cookies are stored for this domain.

**Comment by kmjennison** [comment]
I believe Firebase is the limiting factor because it [doesn't support wildcard subdomains](https://stackoverflow.com/a/54582754). The approach detailed in [this blog post](https://dev.to/brianburton/cross-domain-firebase-authentication-a-simple-approach-337k) might work here (you can change the cookies settings with `next-firebase-auth`), though I haven't thought it through.

If you find an approach that works for you and `next-firebase-auth` is missing some flexibility to support it, please open an issue.

**Reply by viperfx** [reply]
Thanks for the reply @kmjennison - I did read that post- but I am not quite clear how firebase functions fit into this - as we are hosting our app through vercel or some other provider. 

Is there a chance you could share an example of how it could work with next-firebase-auth?

**Reply by kmjennison** [reply]
No, unfortunately I don't have the time to line up a demo for this. My recommendation is to start by getting multidomain Firebase working on its own, then see if/where this library falls short in supporting that setup.
